File: client/archive/ios/BriskVTU/Core/Models/UserProfile.swift
import Foundation

struct UserProfile: Codable, Identifiable {
    let uid: String
    let email: String?
    let phoneNumber: String?
    let firstName: String?
    let lastName: String?
    let stripeCustomerId: String?
    let createdAt: Date?
    let updatedAt: Date?
    
    var id: String { uid }
    
    var avatarUrl: URL? {
        URL(string: "https://api.dicebear.com/7.x/avataaars/png?seed=\(uid)")
    }
    
    enum CodingKeys: String, CodingKey {
        case uid
        case email
        case phoneNumber = "phone_number"
        case firstName = "first_name"
        case lastName = "last_name"
        case stripeCustomerId = "stripe_customer_id"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

File: client/archive/android/app/src/main/java/com/solutioncrafts/BriskVTU/core/models/UserProfile.kt
package com.solutioncrafts.BriskVTU.core.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class UserProfile(
    val id: String,
    val uid: String,
    val email: String? = null,
    @SerialName("phone_number") val phoneNumber: String? = null,
    @SerialName("first_name") val firstName: String? = null,
    @SerialName("last_name") val lastName: String? = null,
    @SerialName("stripe_customer_id") val stripeCustomerId: String? = null,
    @SerialName("created_at") val createdAt: String? = null,
    @SerialName("updated_at") val updatedAt: String? = null
)
